src/_pytest/python.py

```diff
@@ -1134,32 +1132,23 @@ class CallSpec2:
     def setmulti(
         self,
         *,
-        valtypes: Mapping[str, "Literal['params', 'funcargs']"],
         argnames: Iterable[str],
         valset: Iterable[object],
         id: str,
         marks: Iterable[Union[Mark, MarkDecorator]],
         scope: Scope,
         param_index: int,
     ) -> "CallSpec2":
-        funcargs = self.funcargs.copy()
         params = self.params.copy()
         indices = self.indices.copy()
         arg2scope = self._arg2scope.copy()
         for arg, val in zip(argnames, valset):
-            if arg in params or arg in funcargs:
-                raise ValueError(f"duplicate parametrization of {arg!r}")
-            valtype_for_arg = valtypes[arg]
-            if valtype_for_arg == "params":
-                params[arg] = val
-            elif valtype_for_arg == "funcargs":
-                funcargs[arg] = val
-            else:
-                assert_never(valtype_for_arg)
+            if arg in params:
+                raise ValueError(f"duplicate {arg!r}")
```

Comment:
Let's keep the message so as not to undo 1cc58ed67ff9e9c90b6b39154763138779ebceaf.

```suggestion
                raise ValueError(f"duplicate parametrization of {arg!r}")
```